### PR TITLE
Adds more detailed error logging information V3-2239

### PR DESCRIPTION
**Pull request by aligeti**
What: 
Add more info about the nodes when they fail audit
Why:
more error logging detailed info needed for debugging 

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/audit/reporter.go [hunk 1]
```diff
@@ -131,6 +131,7 @@ func (reporter *Reporter) recordAuditFailStatus(ctx context.Context, failedAudit
 		}
 	}
 	if len(failed) > 0 {
+		reporter.log.Debug("Error recording Audit Failed Nodes ", zap.Strings("Node IDs", failed.Strings()))
```
Comment (cam-a) [inline satellite/audit/reporter.go]: It might be worth changing these statements to `log.Error`

File: satellite/audit/reporter.go [hunk 2]
```diff
@@ -148,6 +149,7 @@ func (reporter *Reporter) recordOfflineStatus(ctx context.Context, offlineNodeID
 		}
 	}
 	if len(failed) > 0 {
+		reporter.log.Debug("Error recording Audit Offline Nodes ", zap.Strings("Node IDs", failed.Strings()))
```
Comment (cam-a) [inline satellite/audit/reporter.go]: This logger is named `audit:reporter`, so having the word "Audit" in the message might be unnecessary

File: satellite/audit/reporter.go [hunk 3]
```diff
@@ -199,6 +202,9 @@ func (reporter *Reporter) recordPendingAudits(ctx context.Context, pendingAudits
 		}
 	}
 	if len(failed) > 0 {
+		for _, v := range failed {
+			reporter.log.Debug("Error recording Audit Pending Nodes ", zap.Stringer("Node ID", v.NodeID), zap.String("Path", v.Path))
```
Comment (cam-a) [inline satellite/audit/reporter.go]: I think in general a good message for these logs would be `failed to record x y z`.

File: satellite/audit/reporter.go (hunk 1 above)
Comment (cam-a) [reply satellite/audit/reporter.go]: Oh, actually, I see that the errors themselves are being logged as `Error`. Debug is okay

Comment (cam-a) [reply satellite/audit/reporter.go]: Nevermind, debug is good

Comment (coyle) [reply satellite/audit/reporter.go]: There is a possibility log is nil. I would do a check before calling.

File: satellite/audit/reporter.go [hunk 4]
```diff
@@ -169,6 +171,7 @@ func (reporter *Reporter) recordAuditSuccessStatus(ctx context.Context, successN
 		}
 	}
 	if len(failed) > 0 {
+		reporter.log.Debug("Error recording Audit success Nodes ", zap.Strings("Node IDs", failed.Strings()))
```
Comment (coyle) [inline satellite/audit/reporter.go]: Node IDs -> NodeIDs

File: satellite/audit/reporter.go (hunk 2 above)
Comment (coyle) [inline satellite/audit/reporter.go]: Node IDs -> NodeIDs

File: satellite/audit/reporter.go (hunk 1 above)
Comment (aligeti) [reply satellite/audit/reporter.go]: updated by initializing the log if it is nil

File: satellite/audit/reporter.go (hunk 2 above)
Comment (aligeti) [reply satellite/audit/reporter.go]: updated

Comment (aligeti) [reply satellite/audit/reporter.go]: updated

File: satellite/audit/reporter.go (hunk 4 above)
Comment (aligeti) [reply satellite/audit/reporter.go]: updated

File: satellite/audit/reporter.go (hunk 3 above)
Comment (aligeti) [reply satellite/audit/reporter.go]: updated

File: satellite/audit/reporter.go [hunk 5]
```diff
@@ -36,6 +36,10 @@ type Report struct {
 
 // NewReporter instantiates a reporter
 func NewReporter(log *zap.Logger, overlay *overlay.Cache, containment Containment, maxRetries int, maxReverifyCount int32) *Reporter {
+	if log == nil {
+		log = log.Named("audit:reporter")
```
Comment (egonelbre) [inline satellite/audit/reporter.go]: This shouldn't be done, the logger must come from outside, if it does not then it is a bug.

This check also does not make sense.

Comment (coyle) [reply satellite/audit/reporter.go]: I think it should at minimum be a noop ... I get it's a bug if a logger isn't passed but we shouldn't allow the system to panic

Comment (egonelbre) [reply satellite/audit/reporter.go]: No, this is completely a programmer bug that should never pass tests.

Comment (aligeti) [reply satellite/audit/reporter.go]: it was a comment/suggestion I received from peer review.  I will reopen the branch to update it.

**Pull request merged**